### PR TITLE
Updating analyses/firewx based on Wei's comments

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

This item addresses the following items from Wei's checklist

1)  PDY in config file that is sourced in ecf files.
7) Misleading WARNING in evs_analyses_stats jobs: This section only runs at $vhr=23, otherwise WARNING. This should not be “WARNING” if expected. 
14) Config file name in scripts/stats/cam/jevs_cam_nam_firewxnest_grid2obs_stats_vhr_master.ecf
19) evs_cam_nam_firewxnest_grid2obs_plots failure

1, 14, and 19 are solved by ensuring that the parm file from evs_config is sourced in the exevs script, not the ecf or dev/driver script. True for stats and plots.

7 is resolved by removing the requested WARNING from the output

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Test the firewx stats job.  Run with qsub -v vhr=00.
Test the firewx plots job.  Set COMIN to emc.vpppg
Test one of the analyses stats jobs.  Run with qsub -v vhr=00.

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

File: ecf/scripts/stats/cam/jevs_cam_nam_firewxnest_grid2obs_stats_vhr_master.ecf [hunk 1]
```diff
@@ -59,8 +59,7 @@ export COMOUT=${COMevs}/${STEP}/${COMPONENT}
 export pid=${pid:-$$}
 export outid=${outid:-"LL$job"}
 export DATA=${DATA:-${DATAROOT}/${jobid:?}}
-export config=$HOMEevs/parm/evs_config/${COMPONENT}/config.evs.${COMPONENT}_${MODELNAME}.${envir}
-source $config
+export config=$HOMEevs/parm/evs_config/${COMPONENT}/config.evs.${COMPONENT}_${MODELNAME}.prod
```
Comment (AliciaBentley-NOAA) [inline ecf/scripts/stats/cam/jevs_cam_nam_firewxnest_grid2obs_stats_vhr_master.ecf]: @PerryShafran-NOAA Does this "export config" line need to be in the .ecf script if it's also in scripts/stats/cam/exevs_cam_nam_firewxnest_grid2obs_stats.sh? If you can remove this line from the dev drivers, you can probably remove it from the .ecf scripts too, right? Thanks for checking this. 

CC @ShelleyMelchior-NOAA

Comment (PerryShafran-NOAA) [reply ecf/scripts/stats/cam/jevs_cam_nam_firewxnest_grid2obs_stats_vhr_master.ecf]: Done.

**Comment by ShelleyMelchior-NOAA** [comment]
I have cloned your fork and checked out branch `bugfix/nco_fix`.
I have made changes to run the following jobs from my environment. All processes are currently running.
jevs_cam_nam_firewxnest_grid2obs_stats.sh
jevs_analyses_rtma_ru_grid2obs_stats.sh
jevs_cam_nam_firewxnest_grid2obs_plots.sh

**Comment by ShelleyMelchior-NOAA** [comment]
The rtma_ru stats have completed. My results match emc.vpppg.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_ru_grid2obs_stats.o121619012`
```
$ ll /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/analyses/atmos.20231217/rtma_ru/grid2obs
total 240
-rw-r--r-- 1 shelley.melchior vpppg 121871 Dec 18 22:29 point_stat_rtma2p5_ru_anl_000000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 121856 Dec 18 22:30 point_stat_rtma2p5_ru_ges_010000L_20231217_000000V.stat
```
```
$ ll /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/analyses/atmos.20231217/rtma_ru/grid2obs/*000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 121871 Dec 18 00:01 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/analyses/atmos.20231217/rtma_ru/grid2obs/point_stat_rtma2p
5_ru_anl_000000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 121856 Dec 18 00:02 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/analyses/atmos.20231217/rtma_ru/grid2obs/point_stat_rtma2p
5_ru_ges_010000L_20231217_000000V.stat
```

**Comment by ShelleyMelchior-NOAA** [comment]
The nam firewx stats have completed. My results match emc.vpppg.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_nam_firewxnest_grid2obs_stats_00.o121619007`
```
$ ll /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs
total 144
-rw-r--r-- 1 shelley.melchior vpppg 21691 Dec 18 22:34 point_stat_firewxnest_000000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 15079 Dec 18 22:34 point_stat_firewxnest_060000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 21668 Dec 18 22:34 point_stat_firewxnest_120000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 15123 Dec 18 22:34 point_stat_firewxnest_180000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 21732 Dec 18 22:34 point_stat_firewxnest_240000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 15123 Dec 18 22:34 point_stat_firewxnest_300000L_20231217_000000V.stat
-rw-r--r-- 1 shelley.melchior vpppg 21665 Dec 18 22:34 point_stat_firewxnest_360000L_20231217_000000V.stat
```
```
$ ll /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/*000000V.stat                                                                                                                                                       
-rw-r--r-- 1 emc.vpppg vpppg 21691 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_000000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 15079 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_060000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 21668 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_120000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 15123 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_180000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 21732 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_240000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 15123 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_300000L_20231217_000000V.stat
-rw-r--r-- 1 emc.vpppg vpppg 21665 Dec 18 00:05 /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/cam/atmos.20231217/nam_firewxnest/grid2obs/point_stat_firew
xnest_360000L_20231217_000000V.stat
```

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Everything looks good thus far.  The WARNINGs have been removed, and the stats and plots for firewx ran with no problems.

It did occur to me that even though Wei didn't expressly list this in his doc, I should probably do the same for the analyses component, which I have not done yet.

Please re-test analyses stats and plots.  I can ascertain quickly whether the changes are good.  

@AliciaBentley-NOAA for awareness

**Comment by ShelleyMelchior-NOAA** [comment]
You already saw that the nam firewx plotting finished. But for completeness in this PR:
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/cam/atmos.20231217`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_cam_nam_firewxnest_grid2obs_plots.o121619022`
The job ran for 29 minutes.
`resources_used.walltime = 00:29:10`

**Comment by ShelleyMelchior-NOAA** [comment]
rtma_ru stats has been resubmitted.
analyses plots has been submitted.
logfiles: 
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_ru_grid2obs_stats.o121682914`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o121682937`

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Did you bring in new code before submitting your analyses re-starts?  It looks like you did not.

**Comment by ShelleyMelchior-NOAA** [comment]
> @ShelleyMelchior-NOAA Did you bring in new code before submitting your analyses re-starts? It looks like you did not.

I did, and I will pull in again now that you have sync forked. Maybe that will assure the changes. Then I can submit the two jobs again.

**Comment by ShelleyMelchior-NOAA** [comment]
rtma_ru stats re-submission: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_ru_grid2obs_stats.o121686170`

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA The analyses plot job needed to be restarted as well.  I think that it would be OK though, since I want to get emc.vpppg updated for Lin's ecflow like now.  I'm OK with merging this one as is.

I also added the NET=evs in the SREF ecf script.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA @ShelleyMelchior-NOAA If we could also restart the analyses plot job to complete this test, that would be great! I'm uncomfortable merging untested code for ops to meet an arbitrary deadline. Thanks!

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA Thanks, I appreciate the need to have the tests done.  

@ShelleyMelchior-NOAA We'll have to re-run the plots job.  Apologies if I didn't make that clear earlier, that both stats and plots were updated!

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA For a better test of both stats and plots, please remove the COMOUT directories for both so they are not restarts.  That will make for a cleaner test.

**Comment by ShelleyMelchior-NOAA** [comment]
Existing COMOUT dirs for analyses were removed. New code was pulled in. Re-runs are underway.
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o121690339`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_ru_grid2obs_stats.o121690477`

**Comment by PerryShafran-NOAA** [comment]
The stats job looks perfect!  

The plots job also is good thus far; it has export config in the right place.  We'll just check to see if there is a parsing error.

**Comment by ShelleyMelchior-NOAA** [comment]
I have done the informal code review while we wait for the plots job to completed. The code review looks good.

**Comment by ShelleyMelchior-NOAA** [comment]
> The plots job also is good thus far; it has export config in the right place. We'll just check to see if there is a parsing error.

The plots job is now complete.

**Comment by PerryShafran-NOAA** [comment]
The plots tar file looks identical to the one in emc.vpppg.  No parsing errors or any other errors in the output.

I think we are good on this PR!

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All testing yielded expected results. This PR is approved for merge.

**Pull request merged**